### DISCUSSION TITLE
Delete slected faces

### DISCUSSION THREAD

**Discussion by stromhaug-tommy** [Macad3D/Macad3D#36] [General]
Dear Tobias,

I know that i pushing your patience.
I have managed to solve my previos problems, sort of.

One thing that i am notable to solve is this. I my my simpe viewer i am not abto to select faces.
What i am trying to accomplice is to be able to "remove" faces from solids.

I made a search for OCC in general and found this:

                        aReshape.Remove(myAISContext->SelectedShape());
                        Handle_AIS_InteractiveObject picked;
                        picked = myAISContext->SelectedInteractive();
                        Handle(AIS_Shape) aShape = Handle(AIS_Shape)::DownCast(picked);
                        TopoDS_Shape aNewShape = aReshape.Apply(aShape->Shape());
                        aShape->Set(aNewShape);
                        myAISContext->Redisplay(aShape);

I tried to sove it with this ;

                       TopoDS_Shape detectedShape = Macad.Occt.Helper.Ais.GetDetectedShapeFromContext(AisContext);
                        BRepTools_ReShape aReshape = new BRepTools_ReShape();
                        aReshape.Remove(detectedShape);
                       AIS_InteractiveObject picked = AisContext.SelectedInteractive();
                        AIS_Shape ashape = (AIS_Shape)picked;
                        TopoDS_Shape aNewShape = aReshape.Apply(ashape.Shape());
                        ashape.Set(aNewShape);
                        AisContext.Redisplay(ashape,false);
Unfortnuatly gives me null for picked(AIS_InteractiveObject picked = AisContext.SelectedInteractive();)
Do you please have any idee to helt me ?

**Comment by dipts** [comment]
There's a difference between the _detected_ and the _selected_ shape. Macad has it's own selection handling and thus does only use the detection functions. Try to use `AisContext.SelectedShape()` directly.
